bugbug/bug_features.py

```diff
@@ -905,3 +905,25 @@ class BugType(SingleBugFeature):
 
     def __call__(self, bug, **kwargs):
         return bug["type"]
+
+
+class InitialProduct(SingleBugFeature):
+    def __call__(self, bug, **kwargs):
+        history = bug.get("history", [])
+        if history:
+            for entry in history:
+                for change in entry["changes"]:
+                    if change["field_name"] == "product":
+                        return change["removed"]
+        return None
+
+
+class InitialComponent(SingleBugFeature):
+    def __call__(self, bug, **kwargs):
+        history = bug.get("history", [])
+        if history:
+            for entry in history:
+                for change in entry["changes"]:
+                    if change["field_name"] == "component":
```

Comment:
We should avoid leaking the component name into the training set:
```suggestion
                    if change["field_name"] == "component" and change["removed"] != bug["component"]:
```

In this case, if the first component was the correct one and then moved to another component, it would return the second one instead of the initial one.

Reply:
Done here: b66d5a67e6aff66d1b202bdd2406febbf1070d2f

Currently training the model, will post results once finished.

Reply:
> where the model would move it to an incorrect component, when it actually belongs in the initial component

This will be kind of leaking the classification into the training data, which harms the ability of the model to learn. Also, we do not run the model on bugs that are not in general components, so this is not currently a problem.

Reply:
> The metrics actually appear to be lower

Based on this, I suggest closing the issue as not planned.